rllib/execution/parallel_requests.py

```diff
@@ -279,10 +281,27 @@ def call(
         if fn_kwargs is None:
             fn_kwargs = {}
 
+        def actor_available(a):
+            return (
+                len(self._remote_requests_in_flight[a])
+                < self._max_remote_requests_in_flight
+            )
+
+        num_workers = len(self._all_workers)
+
         if not actor:  # If no actor is specified, use a random actor.
-            actor = self._get_actor_and_mark_availability()
+            for _ in range(num_workers):
+                if actor_available(self._all_workers[self._curr_actor_ptr]):
+                    actor = self._all_workers[self._curr_actor_ptr]
+                    self._curr_actor_ptr = (self._curr_actor_ptr + 1) % num_workers
+                    break
+                self._curr_actor_ptr = (self._curr_actor_ptr + 1) % num_workers
             if not actor:  # No actors available to schedule the request on.
                 return False
+        else:
+            if not actor_available(actor):
+                return False
+            self._curr_actor_ptr = self._actor_to_ptr[actor]
```

Comment:
I actually feel like you don't want to move the _ptr in this case.
sending req to a specific actor is special case, and not part of regular round robin schedule.
then you don't need the _actor_to_ptr map here.

Reply:
fixed